### ISSUE TITLE
Explore ways to improve Machine learning model

### ISSUE THREAD

**Issue by sometheasiekswx** [OnTrack-UG-Squad/speaker-verification#41]
- Here, are the pre-trained models we are using: https://drive.google.com/drive/folders/18h2bmsAWrqoUMsh_FQHDDxp7ioGpcNBa. I found that the ResCNN_triplet_training_checkpoint_265.h5 model we have is better than the ResCNN_softmax_pre_training_checkpoint_102.h5 model. If we can train our own model using what was built in https://github.com/philipperemy/deep-speaker, we could potentially get a better model, especially if we get to use more data to train the model.
- This newer research paper called **GENERALIZED END-TO-END LOSS FOR SPEAKER VERIFICATION** (https://arxiv.org/pdf/1710.10467.pdf) introduces a loss function that makes training our Deep Speaker more efficient and faster. It reduces training time by 60%! This newer paper references the **Deep Speaker: an End-to-End Neural Speaker Embedding System** (https://arxiv.org/pdf/1705.02304.pdf) research paper our model is based on.
- This project (https://github.com/resemble-ai/Resemblyzer) (https://pythonawesome.com/a-python-package-to-analyze-and-compare-voices-with-deep-learning/) has some cool features we should add to our projects like Fake speech detection and Speaker diarization, but more importantly, it has a voice encoder which is based on the paper GENERALIZED END-TO-END LOSS FOR SPEAKER VERIFICATION I mentioned. This project (https://github.com/HarryVolek/PyTorch_Speaker_Verification) has the implementation of the loss function mentioned before that we can use to train our model.
- In the past for model evaluation validation we have used the VCSK corpus as input but could be used for any model training.

**Comment by kugamigh** [comment]
We have decided against working to improve the model, as the return on time invested is not overly important due to the scale we are working with at this point.  If anything changes, we can re-open this issue at a later date.
